usr/sbin/pkcsicsf/pkcsicsf.c

```diff
@@ -191,84 +174,149 @@ int remove_file(char *filename)
     return 0;
 }
 
-static void add_token_config_entry(FILE *fp, const char *key, const char *value)
+static void set_perm(int file)
 {
+    struct group *grp;
+
+    // Set absolute permissions or rw-rw----
+    fchmod(file, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP);
+
+    grp = getgrnam("pkcs11"); // Obtain the group id
+    if (grp) {
+        // set ownership to root, and pkcs11 group
+        if (fchown(file, getuid(), grp->gr_gid) != 0) {
```

Comment:
Why `getuid` and not `0`?

Reply:
well, I guess the comment is wrong, should be '_set ownership to **current user**, and pkcs11 group_'.
This is actually a copy from pkcs_utils.c, but it would make too much trouble to include that module directly due to other unresolved externals, thus the copied routine.....

Reply:
no not necessarily, but the user is part of the pkce11 group. 

However, it is certainly a good question who should be allowed to modify opencryptoki.conf.  Currently its this tool (pkcsicsf) and pkcstok_migrate that programmatically modifies opencryptoki.conf. Currently neither of it is limited to be run by just root. 
It also depends on how the distributions install opencryptoki.conf....

Any idea how to handle this better?

Reply:
OK, changed pkcsicsf as well as pkcstok_migrate to only be usable as root. Also when updating opencryptoki.conf it is then owned by root:root with file mode 644.